File: arch/src/main/java/com/github/lany192/box/utils/ClickUtil.java
package com.github.lany192.box.utils;

/**
 * 防止过快双击
 */
public class ClickUtil {
    private static long lastClickTime;
    private static long INTERVAL = 500;

    public synchronized static boolean isFast() {
        long time = System.currentTimeMillis();
        if (time - lastClickTime < INTERVAL) {
            return true;
        }
        lastClickTime = time;
        return false;
    }
}

File: library/src/main/java/com/github/lany192/box/utils/ClickUtil.kt
package com.github.lany192.box.utils

/**
 * 防止过快双击
 */
object ClickUtil {
    @kotlin.jvm.JvmStatic
    var lastClickTime: Long = 0

    @Synchronized
    @kotlin.jvm.JvmStatic
    fun isFast(): Boolean {
        val time = System.currentTimeMillis()
        if (time - lastClickTime < 500) {
            return true
        }
        lastClickTime = time
        return false
    }
}